### PR TITLE
Update bug report form to use YAML

### PR DESCRIPTION
**Pull request by ybiquitous**
> Which issue, if any, is this issue related to?

Related to https://github.com/stylelint/.github/issues/5

> Is there anything in the PR that needs further explanation?

You can view the new form on [`.github/ISSUE_TEMPLATE/REPORT_A_BUG.yml`](https://github.com/stylelint/stylelint/blob/new-format-of-bug-report/.github/ISSUE_TEMPLATE/REPORT_A_BUG.yml).
This is based on the [issue form on `stylelint/.github`](https://github.com/stylelint/.github/blob/eac961d931bb857d86791be1aacc409fc05fc8f5/.github/ISSUE_TEMPLATE/REPORT_A_BUG.yml).

See also:
- The current [issue template](https://github.com/stylelint/stylelint/blob/85cc19c73bc7751edca9745548d00e01252cd1c7/.github/ISSUE_TEMPLATE/REPORT_A_BUG.md)
- The [GitHub documentation](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-githubs-form-schema)

Note:
I'm not sure, but we may be able to delete the remaining issue templates `REPORT_A_DOCS_ISSUE.md` and `REQUEST_A_FEATURE.md`. Because I think they should have similar content as [ones on `stylelint/.github`](https://github.com/stylelint/.github/tree/eac961d931bb857d86791be1aacc409fc05fc8f5/.github/ISSUE_TEMPLATE).

### CODE REVIEW

**Comment by ybiquitous** [comment]
@stylelint/owners 

I've created the new bug report form, but I'm not sure if the form is good enough. 😓 
If you have ideas to make better contributions, please let me know. 🙏🏼

**Comment by mattxwang** [comment]
Not an owner, but just wanted to comment and say that I really like this! Great work @ybiquitous!

**Comment by ybiquitous** [comment]
@mattxwang No worry, any feedback is welcome! 😄

File: .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
+name: '🐛 Report a bug'
+description: 'Is something not working as you expect?'
+labels: []
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Thanks for taking the time to file a bug report!
```
Comment (jeddy3) [inline .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml:8]:
```suggestion
        Thanks for taking the time to file a bug report!
        
        Before posting, please check that the bug hasn't already been:
        -  fixed in the next release (https://github.com/stylelint/stylelint/blob/main/CHANGELOG.md)
        -  discussed previously (https://github.com/stylelint/stylelint/search)

       You can help us fix the bug more quickly by:
       1. Figuring out what needs doing and proposing it.
       2. Writing the code and submitting a PR once the bug is confirmed.
```

Let's add this blurb back in as it:
- reduces duplicates
- encourages people to get involved

File: .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  - type: textarea
+    id: reproduce-bug
+    attributes:
+      label: 'What steps are needed to reproduce the bug?'
+      placeholder: |
+        e.g. "CLI with `stylelint '**/*.css' --config myconfig.json`"
```
Comment (jeddy3) [inline .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml:14]:
```suggestion
```

Let's start with empty for consistency with the other templates and see what we end up with.

File: .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      placeholder: |
+        e.g. "Yes, it's related to SCSS nested properties."
+    validations:
+      required: false
+  - type: textarea
+    id: expected-behavior
+    attributes:
+      label: 'What did you expect to happen?'
+      placeholder: |
+        e.g. "No warnings to be flagged."
```
Comment (jeddy3) [inline .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml:60]:
```suggestion
        e.g. "No problems to be reported."
```

Align with new language.

File: .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    validations:
+      required: true
+  - type: textarea
+    id: actual-behavior
+    attributes:
+      label: 'What actually happened?'
+      description: 'For example, what warnings or errors did you get?'
+      placeholder: |
+        e.g. "The following warnings were flagged:"
```
Comment (jeddy3) [inline .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml:69]:
```suggestion
        e.g. "The following problems were reported:"
```

File: .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    id: stylelint-version
+    attributes:
+      label: 'Which version of Stylelint are you using?'
+      placeholder: 'e.g. `14.0.0`'
+    validations:
+      required: false
+  - type: input
+    id: non-standard-syntax
+    attributes:
+      label: 'Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?'
```
Comment (jeddy3) [inline .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml:50]:
```suggestion
      label: 'Does the bug relate to non-standard syntax?'
      description: "For example, a construct from SCSS, Less or CSS-in-JS etc.)
```

File: .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    id: stylelint-rule
+    attributes:
+      label: 'Which rule, if any, is the bug related to?'
+      placeholder: 'e.g. `number-leading-zero`'
+    validations:
+      required: false
+  - type: textarea
+    id: stylelint-configuration
+    attributes:
+      label: 'What Stylelint configuration is needed to reproduce the bug?'
```
Comment (jeddy3) [inline .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml:27]:
```suggestion
      label: 'What Stylelint configuration is needed to reproduce the bug?'
      description: Please provide the smallest reproducible example.
```

File: .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+        e.g. "CLI with `stylelint '**/*.css' --config myconfig.json`"
+    validations:
+      required: true
+  - type: input
+    id: stylelint-rule
+    attributes:
+      label: 'Which rule, if any, is the bug related to?'
+      placeholder: 'e.g. `number-leading-zero`'
+    validations:
+      required: false
```
Comment (jeddy3) [inline .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml:23]: Let's lose the `stylelint-rule` field as it's redundant if we ask for the smallest reproducible config.

File: .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+  - type: input
+    id: non-standard-syntax
+    attributes:
+      label: 'Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?'
+      placeholder: |
+        e.g. "Yes, it's related to SCSS nested properties."
+    validations:
+      required: false
+  - type: textarea
```
Comment (jeddy3) [inline .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml:55]: Let's move `expected-behaviour` and `actual-behavior` to after `stylelint-version`.

But let's add a section after `stylelint-configuration` first:

- label: How do you run Stylelint?
- description: For example, CLI, Node,js API or PostCSS plugin.
- placeholder: e.g. "CLI with `stylelint "**/*.css" --config myconfig.json`"

So it goes:
- reproduce-bug
- stylelint-configuration
- run (new section)
- stylelint-version
- expected-behaviour
- actual-behaviour
- non-standard-syntax

So it follows the normal flow of:
- steps
- expected
- actual

But we're pulling out and specifically asking for the config, how it's being run and Stylelint version from the steps.

The final question, `non-standard-syntax`, is there to prepare users for when we say that Stylelint is designed for CSS, and the suggestion fixed well either be:
- we update our heuristics to ignore the syntax in Stylelint
- suggest they raise an issue in a 3rd party config or custom syntax

**Review by jeddy3 — CHANGES_REQUESTED** [review]
@ybiquitous Thanks, it's off to a good start.

I've made suggestions, including removing, adding and reordering fields.

> I'm not sure, but we may be able to delete the remaining issue templates REPORT_A_DOCS_ISSUE.md and REQUEST_A_FEATURE.md. Because I think they should have similar content as ones on stylelint/.github.

If the new YAML versions of those two templates will filter through from `.github` then that would be fab, and we don't need to recreate them here.

File: .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml (hunk 1 above)
Comment (ybiquitous) [reply .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml:8]: Done! e516da71

File: .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml (hunk 7 above)
Comment (ybiquitous) [reply .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml:29]: Do you mean that we should delete this field?

File: .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml (hunk 6 above)
Comment (ybiquitous) [reply .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml:33]: Done. 566be1d5

File: .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml (hunk 5 above)
Comment (ybiquitous) [reply .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml]: Done! 632cf876

File: .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml (hunk 7 above)
Comment (ybiquitous) [reply .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml]: Done. c128ec66

I think the `stylelint-rule` field is no longer needed from [the comment](https://github.com/stylelint/stylelint/pull/5678/files#r739686521).

File: .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml (hunk 8 above)
Comment (ybiquitous) [reply .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml:56]: Sounds good! Fixed: fd3eb80b, 1755f793

**Comment by ybiquitous** [comment]
@jeddy3 Thanks a lot! I've fixed all reviews. 👍🏼

File: .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+labels: []
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Thanks for taking the time to file a bug report!
+
+        Before posting, please check that the bug hasn't already been:
+        * fixed in the next release (https://github.com/stylelint/stylelint/blob/main/CHANGELOG.md)
+        * discussed previously (https://github.com/stylelint/stylelint/search)
```
Comment (jeddy3) [inline .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml:12]:
```suggestion
        * [fixed in the next release](https://github.com/stylelint/stylelint/blob/main/CHANGELOG.md)
        * [discussed previously](https://github.com/stylelint/stylelint/search)
```

Let's use markdown links to clean this up a little.

File: .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+
+        ```json
+        {
+          "rules": {
+            "number-leading-zero": "always"
+          }
+        }
+        ```
+    validations:
+      required: false
```
Comment (jeddy3) [inline .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml:39]:
```suggestion
      required: true
```

Let's this, and the other fields that pull out specific information from the steps, required.

File: .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+  - type: input
+    id: stylelint-run
+    attributes:
+      label: 'How do you run Stylelint?'
```
Comment (jeddy3) [inline .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml:43]:
```suggestion
      label: 'How did you run Stylelint?'
```

Let's make it specific to this bug report.

File: .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+      description: 'For example, CLI, Node,js API or PostCSS plugin.'
+      placeholder: |
+        e.g. "CLI with `stylelint "**/*.css" --config myconfig.json`"
```
Comment (jeddy3) [inline .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml:46]:
```suggestion
        CLI with `stylelint "**/*.css" --config myconfig.json`
```

Let's remove the "e.g." and quotes as they're redundant in placeholders(, but they aren't before). It'll mean more of the placeholder content it show too.

File: .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+    validations:
+      required: false
```
Comment (jeddy3) [inline .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml:48]:
```suggestion
      required: true
```

File: .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml [hunk 14]
```diff
@@ -0,0 +1,86 @@
...
+  - type: input
+    id: stylelint-version
+    attributes:
+      label: 'Which version of Stylelint are you using?'
+      placeholder: 'e.g. `14.0.0`'
```
Comment (jeddy3) [inline .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml:53]:
```suggestion
      placeholder: '`14.0.0`'
```

File: .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml [hunk 15]
```diff
@@ -0,0 +1,86 @@
...
+    validations:
+      required: false
+  - type: textarea
+    id: expected-behavior
+    attributes:
+      label: 'What did you expect to happen?'
+      placeholder: |
+        e.g. "No problems to be reported."
```
Comment (jeddy3) [inline .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml:61]:
```suggestion
        No problems to be reported.
```

File: .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml [hunk 16]
```diff
@@ -0,0 +1,86 @@
...
+    validations:
+      required: true
+  - type: textarea
+    id: actual-behavior
+    attributes:
+      label: 'What actually happened?'
+      description: 'For example, what warnings or errors did you get?'
+      placeholder: |
+        e.g. "The following problems were reported:"
```
Comment (jeddy3) [inline .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml:70]:
```suggestion
        The following problems were reported:
```

File: .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml [hunk 17]
```diff
@@ -0,0 +1,86 @@
...
+        ```
+    validations:
+      required: true
+  - type: input
+    id: non-standard-syntax
+    attributes:
+      label: 'Does the bug relate to non-standard syntax?'
+      description: 'For example, a construct from SCSS, Less or CSS-in-JS.'
+      placeholder: |
+        e.g. "Yes, it's related to SCSS nested properties."
```
Comment (jeddy3) [inline .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml:84]:
```suggestion
        Yes, it's related to SCSS nested properties.
```

File: .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml [hunk 18]
```diff
@@ -0,0 +1,86 @@
+name: '🐛 Report a bug'
+description: 'Is something not working as you expect?'
+labels: []
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Thanks for taking the time to file a bug report!
```
Comment (jeddy3) [inline .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml:8]:
```suggestion
        Thank you for taking the time to file a bug report!
```

Let's tweak this it's more in the active voice.

File: .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml [hunk 19]
```diff
@@ -0,0 +1,86 @@
...
+        e.g. "CLI with `stylelint "**/*.css" --config myconfig.json`"
+    validations:
+      required: false
+  - type: input
+    id: stylelint-version
+    attributes:
+      label: 'Which version of Stylelint are you using?'
+      placeholder: 'e.g. `14.0.0`'
+    validations:
+      required: false
```
Comment (jeddy3) [inline .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml:55]:
```suggestion
      required: true
```

**Review by jeddy3 — CHANGES_REQUESTED** [review]
@ybiquitous Thanks for making the changes! I think it's almost there.

I've requested a handful of tweaks.

File: .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml [hunk 20]
```diff
@@ -0,0 +1,86 @@
...
+    attributes:
+      value: |
+        Thanks for taking the time to file a bug report!
+
+        Before posting, please check that the bug hasn't already been:
+        * fixed in the next release (https://github.com/stylelint/stylelint/blob/main/CHANGELOG.md)
+        * discussed previously (https://github.com/stylelint/stylelint/search)
+
+        You can help us fix the bug more quickly by:
+        1. Figuring out what needs doing and proposing it.
```
Comment (jeddy3) [inline .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml:15]:
```suggestion
        1. Figuring out what needs doing and proposing it in the issue.
```

File: .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml [hunk 21]
```diff
@@ -0,0 +1,86 @@
...
+        2. Writing the code and submitting a PR once the bug is confirmed.
```
Comment (jeddy3) [inline .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml:16]:
```suggestion
        2. After the bug is confirmed, [contributing](https://github.com/stylelint/stylelint/blob/main/CONTRIBUTING.md) a pull request.
```

File: .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml [hunk 22]
```diff
@@ -0,0 +1,86 @@
...
+    validations:
+      required: true
+  - type: textarea
+    id: stylelint-configuration
+    attributes:
+      label: 'What Stylelint configuration is needed to reproduce the bug?'
+      description: 'Provide the smallest reproducible example.'
+      placeholder: |
+        e.g.
+
```
Comment (ybiquitous) [inline .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml:30]: @jeddy3 Is this `e.g.` also no longer needed?

```suggestion
```

File: .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml (hunk 18 above)
Comment (ybiquitous) [reply .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml]: Sounds good! I'll update also the issue forms via <https://github.com/stylelint/.github/pull/8>.

File: .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml (hunk 22 above)
Comment (jeddy3) [reply .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml:30]: Yeap, that goes too.

File: .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml [hunk 23]
```diff
@@ -0,0 +1,84 @@
...
+labels: []
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Thank you for taking the time to file a bug report!
+
+        Before posting, please check that the bug hasn't already been:
+        * [fixed in the next release](https://github.com/stylelint/stylelint/blob/main/CHANGELOG.md)
+        * [discussed previously](https://github.com/stylelint/stylelint/search)
```
Comment (jeddy3) [inline .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml:12]:
```suggestion
        * [discussed previously](https://github.com/search/advanced?r=stylelint%2Fstylelint&type=Issues)
```

The page at the previous link must of changed at some point as it's not very useful now. This new one has a search bar at the top (and will search only issues).

**Review by jeddy3 — CHANGES_REQUESTED** [review]
One last suggestion, and one query:

Shall we add an (optional) field at the bottom to gives users a place to "Figuring out what needs doing and proposing it in the issue."?

For example

```
- label: Proposal to fix the bug
- description: If you know what needs doing, please propose it below.
- validations:
      required: false
```

File: .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml (hunk 23 above)
Comment (ybiquitous) [reply .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml]: Looks good!

FYI. I think <https://github.com/stylelint/stylelint/search?type=issues> is also good. Anyway, the search link can be improved. 😄

**Comment by ybiquitous** [comment]
> Shall we add an (optional) field at the bottom to gives users a place to "Figure out what needs doing and propose it in the issue."?

Sounds interesting. I've added it via 119e2aff.

**Review by jeddy3 — APPROVED** [review]
Fantastic, thank you!

We can merge this now as it's a leap in the right direction. We can see how it goes over the next couple of weeks, and refine it if neccessary.

**Pull request merged**